api/analyzers/typescript/analyzer.py

```
@@ -0,0 +1,119 @@
+
+import tree_sitter_typescript as tstypescript
+from tree_sitter import Language, Node
+
+import logging
+logger = logging.getLogger('code_graph')
+
+class TypeScriptAnalyzer(AbstractAnalyzer):
+    def __init__(self) -> None:
+        super().__init__(Language(tstypescript.language_typescript()))
+
+    def add_dependencies(self, path: Path, files: list[Path]):
```

Ruff (0.15.5)

[warning] Unused method argument: `files`

(ARG002)